client/src/nl/sensorlab/videowall/animation/baseanimations/flocking/Boid.java

```diff
@@ -0,0 +1,190 @@
+package nl.sensorlab.videowall.animation.baseanimations.flocking;
+
+import java.util.ArrayList;
+
+import nl.sensorlab.videowall.animation.BaseAnimation;
+import processing.core.PApplet;
+import processing.core.PGraphics;
+import processing.core.PVector;
+
+public class Boid {
+
+	private PVector position;
+	private PVector velocity;
+	private PVector acceleration;
+
+
+	public float radius = 0.5f;
+	public float maxSpeed = 0.35f;
+	public float maxForce = 0.355f;
+	public float neighbourDistance = 20;
+
+	int color;
+
+	public Boid(float x ,float y, int color) {
+		this.position = new PVector(x, y);
+		this.acceleration = new PVector(0,0);
+		this.color = color;
+
+		float angle = (float)(Math.random() * (Math.PI * 2));
+		this.velocity = new PVector((float)Math.cos(angle), (float)Math.sin(angle)); 
+	}
+
+	public void update(ArrayList<Boid> boids) {
+		flock(boids);
+
+		// Update velocity
+		velocity.add(acceleration);
+
+		// Limit speed
+		velocity.limit(maxSpeed);
+		position.add(velocity);
+
+		// Reset accelertion to 0 each cycle
+		acceleration.mult(0);
+
+		// Keep the boid within the window
+		keepInBounds();
+	}
+
+	private void flock(ArrayList<Boid> boids) {	
+		PVector sep = separate(boids);   // Separation
```

Comment:
For performance it would be a lot better if there would be one loop over all boids in the flock, rather than doing it three times for separation, alignment and cohesion